lib/utils/diff.js

```diff
@@ -0,0 +1,59 @@
+var deepEqual = require('deep-equal')
+var arrayToMap = require('./array-to-map')
+
+// all changes types
+var types = arrayToMap(['CREATED', 'CHANGED', 'REMOVED'])
+
+/**
+ * Get differences between two objects
+ *
+ * @param  {Object} root
```

Comment:
Probably just a personal preference, but I'm not a fan of `root` as a parameter name. It makes me (and my Node IDE bits) think that it's `env.root`